mlir/include/mlir/Dialect/Rock/IR/RockAttrDefs.td

```diff
@@ -200,6 +201,7 @@ def Rock_GeneralGemmParamsAttr : Rock_Attr<"GeneralGemmParams", [RockTuningParam
     - kpack: The number of values of k to pack contiguously into the shared buffer
   }];
   let parameters = (ins
+    "uint32_t":$blockSize,
```

Comment:
Don't put this here. We already have blockSize as a separate attribute.

Alternatively, make this change for xdlops as well, and get rid of the blockSize attribute.

Reply:
(Honestly, either one's fine, and putting block size in the big params struct might not be a bad cleanup ... just do it uniformly across xdlops and non-xdlops)

Reply:
I can't easily agree to this.

blocksize is turned out to be a proper tuning parameter, needs to be there.
general/xdlops are already quite different ops, we don't need to have uniform tuning parameter over different ops.

Also, if we get blocksize into the xdlopsGemm's params set, that will break uniformity in tuning flow. 
Either user needs to provide correctly calculated blocksize in the tuning params or we need to ignore the given number.

For now, it only duplicates the value in op.attr and param since AffixTuningParameter

I'd suggest to revisit this problem when we remove perfconfig string in near future.

Reply:
And I can't agree to that duplication, they might get out of sync

Reply:
I am, however, willing to go for this:
- On the high-level ops (conv, gemm, etc., the ones that don't know if they're xdlops or not), rename `blockSize` to something like `derivedBlockSize` - and possibly do the same to `gridSize` (to `derivedGridSize`)
- In the non-xdlops flow, setting `derivedBlockSize` is an error (update `::verify`, either on the interfaces or on the ops themselves, to check this), and add `blockSize` to the tuning parameters like you did
- Add a `RockGemmWrapperInterface` method `Optional<uint32_t> getBlockSize()` that returns the blockSize from general tuning parameters if those are available and goes for `derivedBlockSize` otherwise.

Reply:
(or maybe don't need the interface)
- Delete blockSize attribute on gridwise_gemm, blockwise_gemm, ...
- **Don't** delete or rename blockSize on gridwise_gemm_v2, blockwise_gemm_v2, ...